wear/src/main/java/io/homeassistant/companion/android/home/HomeActivity.kt

```diff
@@ -91,4 +89,159 @@ class HomeActivity : AppCompatActivity(), HomeView {
         startActivity(intent)
         finish()
     }
+
+    @Composable
+    private fun LoadHomePage() {
+
+        val entities = runBlocking {
+            presenter.getEntities()
```

Comment:
This should be passed into the composable not called within. 
 https://developer.android.com/jetpack/compose/state#state-in-composables

By doing that you should be able to detach the loading of the entities from the initial creation of the view.   That should help with the slowness in loading the view.

Reply:
Not sure I notice it loading any faster, subsequent loads are faster though so maybe its that initial load will be slow?

Reply:
This is now taken care of with the new view model